src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -0,0 +1,54 @@
+package com.google.flourbot.execution;
+
+import com.google.flourbot.datastorage.DataStorage;
+import com.google.flourbot.datastorage.FirebaseDataStorage;
+import com.google.flourbot.entity.EntityModule;
+import com.google.flourbot.entity.EntityModuleImplementation;
+import com.google.flourbot.entity.Macro;
+import com.google.flourbot.entity.action.Action;
+import com.google.flourbot.entity.action.ActionType;
+
+import java.util.Optional;
+
+// The Logic class of the server
+public class MacroExecutionModuleImplementation {
+
+  private final EntityModule entityModule;
+
+  private MacroExecutionModuleImplementation(EntityModule entityModule) {
+    this.entityModule = entityModule;
+  }
+
+  public static MacroExecutionModuleImplementation initalizeServer() {
+    DataStorage dataStorage = new FirebaseDataStorage();
+    EntityModule entityModule = new EntityModuleImplementation(dataStorage);
+
+    return new MacroExecutionModuleImplementation(entityModule);
+  }
+
+  public String execute(String userEmail, String message) {
+
+    String macroName = message.split(" ")[0];
+
+    Optional<Macro> optionalMacro = entityModule.getMacro(userEmail, macroName);
+    if (!optionalMacro.isPresent()) {
+      return "No macro of name: " + macroName + " found";
+    }
+
+    Action action = optionalMacro.get().getAction();
+    ActionType actionType = action.getActionType();
+
+    switch (actionType) {
+      case SHEET_APPEND:
+        // Document writer stuff
+
+        break;
+      default:
+        throw new IllegalStateException(
+            "Action type named: " + actionType.toString() + "is not implemented yet!");
+    }
+
+    // TODO: Return a response object
+    return "Sucessfully executed";
+  }
+}
```

Comment:
This pull request have some automatic formatter changes that are not important. Major changes are 1. "Using an initalizer function to construct all the modules we need" and 2. Converting checked exception into the unchecked version where appropriate.